tests/components/bsblan/conftest.py

```diff
@@ -71,3 +77,44 @@ async def init_integration(
     await hass.async_block_till_done()
 
     return mock_config_entry
+
+
+@pytest.fixture
+def climate(hass: HomeAssistant, mock_config_entry, mock_bsblan) -> BSBLANClimate:
+    """Set up the BSBLan climate entity for testing."""
+    mock_device_registry(hass)
+    coordinator = BSBLanUpdateCoordinator(hass, mock_config_entry, mock_bsblan)
+    coordinator.config_entry = mock_config_entry
+    coordinator.data = BSBLanCoordinatorData(state=mock_bsblan.state.return_value)
+    data = BSBLanData(
+        client=mock_bsblan,
+        coordinator=coordinator,
+        device=mock_bsblan.device.return_value,
+        info=mock_bsblan.info.return_value,
+        static=mock_bsblan.static_values.return_value,
+    )
+    hass.data.setdefault("bsblan", {})[mock_config_entry.entry_id] = data
+    return BSBLANClimate(data)
+
+
+@pytest.fixture
+def climate_fahrenheit(
+    hass: HomeAssistant, mock_config_entry, mock_bsblan
+) -> BSBLANClimate:
+    """Set up the BSBLan climate entity with Fahrenheit temperature unit."""
+    mock_device_registry(hass)
+    coordinator = BSBLanUpdateCoordinator(hass, mock_config_entry, mock_bsblan)
+    coordinator.config_entry = mock_config_entry
+    coordinator.data = BSBLanCoordinatorData(state=mock_bsblan.state.return_value)
+    # override min_temp unit to Fahrenheit in mock_bsblan static values
+    mock_bsblan.static_values.return_value.min_temp.unit = "°F"
+
+    data = BSBLanData(
+        client=mock_bsblan,
+        coordinator=coordinator,
+        device=mock_bsblan.device.return_value,
+        info=mock_bsblan.info.return_value,
+        static=mock_bsblan.static_values.return_value,
+    )
+    hass.data.setdefault("bsblan", {})[mock_config_entry.entry_id] = data
+    return BSBLANClimate(data)
```

Comment:
This is still not what we want. Please check an integration like `airgradient` and `flexit_bacnet` for a good example. We want the integration to act like a real life example

Reply:
not yet 100% now. I'll have another look how I can increase the coverage with this method.

Reply:
Check the air gradient tests. we can setup the integration, call a service and then see that the library is called in a right way